### ISSUE TITLE
iOS: Stuck on cloning: GitObjectNotFound

### ISSUE THREAD

**Issue by mcalus3** [GitJournal/GitJournal#630]
Hi,

I can't clone my existing GitHub repo, nor newly created GitHub repo. On existing repo i have an error during cloning:GitObjectNotFound: 6acf0[...]

On creating a new repo i have an error GitBranchAlreadyExists: mainI am using the iOS app, already sent the bug report about that.

Does anyone stumbled upon similar issue and know how i can work around that?

Here is an excerpt from logs:
```
{"t":1652260165206,"l":"i","msg":"RepoPath: /var/mobile/Containers/Data/Application/B9D14E53-49D8-4425-BA8F-A4E79C9DFD8B/Documents/journal"}
{"t":1652260169250,"l":"i","msg":"Got default branch: master"}
{"t":1652260169250,"l":"i","msg":"Using remote branch: master"}
{"t":1652260169256,"l":"i","msg":"Repo has the following branches: [master, main]"}
{"t":1652260169257,"l":"i","msg":"Local branches [master, main]"}
{"t":1652260169257,"l":"i","msg":"Completing - localBranch: master"}
{"t":1652260169258,"l":"i","msg":"Current branch is not the only branch"}
{"t":1652260169258,"l":"d","msg":"Current Branch: main"}
{"t":1652260169258,"l":"d","msg":"Branch: master"}
{"t":1652260169275,"l":"e","msg":"Failed to clone","ex":"GitObjectNotFound: 6acf0206e739c794c0479580a18f2b6d011ed888","stack":[{"uri":"package:dart_git/utils/result.dart","line":13,"member":"new Result.fail","isCore":false,"library":"package:dart_git/utils/result.dart","location":"package:dart_git/utils/result.dart 13","package":"dart_git"},{"uri":"package:dart_git/storage/object_storage_fs.dart","line":55,"member":"ObjectStorageFS.read","isCore":false,"library":"package:dart_git/storage/object_storage_fs.dart","location":"package:dart_git/storage/object_storage_fs.dart 55","package":"dart_git"},{"uri":"package:dart_git/storage/object_storage_exception_catcher.dart","line":13,"member":"ObjectStorageExceptionCatcher.read.<fn>","isCore":false,"library":"package:dart_git/storage/object_storage_exception_catcher.dart","location":"package:dart_git/storage/object_storage_exception_catcher.dart 13","package":"dart_git"},{"uri":"package:dart_git/utils/result.dart","line":93,"member":"catchAllSync","isCore":false,"library":"package:dart_git/utils/result.dart","location":"package:dart_git/utils/result.dart 93","package":"dart_git"},{"uri":"package:dart_git/storage/object_storage_exception_catcher.dart","line":13,"member":"ObjectStorageExceptionCatcher.read","isCore":false,"library":"package:dart_git/storage/object_storage_exception_catcher.dart","location":"package:dart_git/storage/object_storage_exception_catcher.dart 13","package":"dart_git"},{"uri":"package:dart_git/storage/object_storage_extensions.dart","line":45,"member":"ObjectStorageExtension.readBlob","isCore":false,"library":"package:dart_git/storage/object_storage_extensions.dart","location":"package:dart_git/storage/object_storage_extensions.dart 45","package":"dart_git"},{"uri":"package:dart_git/checkout.dart","line":139,"member":"Checkout._checkoutBranch","isCore":false,"library":"package:dart_git/checkout.dart","location":"package:dart_git/checkout.dart 139","package":"dart_git"},{"uri":"package:dart_git/checkout.dart","line":98,"member":"Checkout.checkoutBranch.<fn>","isCore":false,"library":"package:dart_git/checkout.dart","location":"package:dart_git/checkout.dart 98","package":"dart_git"},{"uri":"package:dart_git/utils/result.dart","line":93,"member":"catchAllSync","isCore":false,"library":"package:dart_git/utils/result.dart","location":"package:dart_git/utils/result.dart 93","package":"dart_git"},{"uri":"package:dart_git/checkout.dart","line":98,"member":"Checkout.checkoutBranch","isCore":false,"library":"package:dart_git/checkout.dart","location":"package:dart_git/checkout.dart 98","package":"dart_git"},{"uri":"package:dart_git/git_async.dart","line":411,"member":"_processCommand","isCore":false,"library":"package:dart_git/git_async.dart","location":"package:dart_git/git_async.dart 411","package":"dart_git"},{"uri":"package:dart_git/git_async.dart","line":374,"member":"_isolateMain.<fn>","isCore":false,"library":"package:dart_git/git_async.dart","location":"package:dart_git/git_async.dart 374","package":"dart_git"}]}
{"t":1652260169277,"l":"d","msg":"Event.GitHostSetupGitCloneError","p":{"error":"GitObjectNotFound: 6acf0206e739c794c0479580a18f2b6d011ed888"}}
[...]
{"t":1652261288256,"l":"i","msg":"RepoPath: /var/mobile/Containers/Data/Application/B9D14E53-49D8-4425-BA8F-A4E79C9DFD8B/Documents/journal"}
{"t":1652261291013,"l":"w","msg":"Could not fetch git Default Branch","ex":"GitException: Unknown Error"}
{"t":1652261291015,"l":"e","msg":"Failed to guess RemoteHead. Returning `main`"}
{"t":1652261291015,"l":"i","msg":"Using remote branch: main"}
{"t":1652261291021,"l":"i","msg":"Repo has the following branches: [master, main]"}
{"t":1652261291022,"l":"i","msg":"Local branches [master, main]"}
{"t":1652261291022,"l":"i","msg":"Completing - localBranch diff remote: master main"}
{"t":1652261291024,"l":"e","msg":"Failed to clone","ex":"GitBranchAlreadyExists: main","stack":[{"uri":"package:dart_git/utils/result.dart","line":13,"member":"new Result.fail","isCore":false,"library":"package:dart_git/utils/result.dart","location":"package:dart_git/utils/result.dart 13","package":"dart_git"},{"uri":"package:dart_git/git.dart","line":311,"member":"GitRepository.createBranch","isCore":false,"library":"package:dart_git/git.dart","location":"package:dart_git/git.dart 311","package":"dart_git"},{"uri":"package:dart_git/git_async.dart","line":401,"member":"_processCommand","isCore":false,"library":"package:dart_git/git_async.dart","location":"package:dart_git/git_async.dart 401","package":"dart_git"},{"uri":"package:dart_git/git_async.dart","line":374,"member":"_isolateMain.<fn>","isCore":false,"library":"package:dart_git/git_async.dart","location":"package:dart_git/git_async.dart 374","package":"dart_git"}]}
{"t":1652261291026,"l":"d","msg":"Event.GitHostSetupGitCloneError","p":{"error":"GitBranchAlreadyExists: main"}}
{"t":1652261299483,"l":"d","msg":"Starting RepoSelector"}
{"t":1652261303307,"l":"d","msg":"Event.DrawerSettings"}
{"t":1652262690437,"l":"d","msg":"Event.DrawerSettings"}
```

**Comment by vHanda** [comment]
Confirmed. I'll take a look tomorrow morning. It's the only item in my GitJournal todo list for tomorrow.

**Comment by vHanda** [comment]
I've made progress, but haven't managed it fix it.

Maybe I'll get some time tonight, otherwise tomorrow.

**Comment by vHanda** [comment]
Fixed with `1.85`

**Comment by mcalus3** [comment]
Thanks, it works now :)
